elfo-core/src/init.rs

```diff
@@ -85,7 +109,20 @@ pub async fn start(topology: Topology) {
 
 /// The same as `start()`, but returns an error rather than panics.
 pub async fn try_start(topology: Topology) -> Result<()> {
-    let res = do_start(topology, termination).await;
+    let res = do_start(topology, false, termination).await;
+
+    if res.is_err() {
+        // XXX: give enough time to the logger.
+        sleep(Duration::from_millis(500)).await;
+    }
+
+    res
+}
+
+/// Starts node in "check only" mode. Entrypoints are started, then the system
+/// is immediately gracefully terminated.
+pub async fn check_only(topology: Topology) -> Result<()> {
+    let res = do_start(topology, true, do_termination).await;
 
     if res.is_err() {
         // XXX: give enough time to the logger.
```

Comment:
The logger won't be started in the only_check mode, so you don't need to do this hack here.